### PR TITLE
Fix annotation docs to advertise string converters.

### PR DESCRIPTION
**Pull request by NightlyNexus**

### CODE REVIEW

File: retrofit/src/main/java/retrofit2/http/Field.java [hunk 1]
```diff
@@ -15,17 +15,22 @@
...
+import retrofit2.Retrofit;
 
 import static java.lang.annotation.ElementType.PARAMETER;
 import static java.lang.annotation.RetentionPolicy.RUNTIME;
 
 /**
  * Named pair for a form-encoded request.
  * <p>
- * Values are converted to strings using {@link String#valueOf(Object)} and then form URL encoded.
+ * Values are converted to strings using {@link Retrofit#stringConverter(Type, Annotation[])}
```
Comment (digitalbuddha) [inline retrofit/src/main/java/retrofit2/http/Field.java]: converted to ~~strings~~Strings?

File: retrofit/src/main/java/retrofit2/http/Field.java [hunk 2]
```diff
@@ -15,17 +15,22 @@
...
+ * (or {@link Object#toString()}, if no matching string Converter is installed)
```
Comment (digitalbuddha) [inline retrofit/src/main/java/retrofit2/http/Field.java]: same, `String` should be class case?

Comment (JakeWharton) [inline retrofit/src/main/java/retrofit2/http/Field.java]: I'd prefer if "Converter" was lowercased instead. Saying "string converter" is fine.

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**